public/js/helpers/zoom_helper.js

```diff
@@ -121,7 +121,8 @@ export default class Zoom {
     for (let k in zoomMap) {
       let v = zoomMap[k]
       if (v === 0) continue
-      if (v / scale === range) return k
+      // support an error of ±1 due to precision loss while dividing.
+      if (Math.abs(v / scale - range) < 1) return k
```

Comment:
Good call. Note that this only works when the range is large, >> 1, which it is for time and block height. This will break if ever used for a chart with say range 0 < x < 1. 

A more flexible solution would be to check if the node is within say +/-1 **%** of the visible window, so this could be 

```
let scaled = v / scale
if (Math.abs((scaled - range) / scaled) < 0.01) return k 
```
*untested

Reply:
I think the ±1% check is way too huge and could distort information more than ±1.

Consider this:

       A day has a range of 288 blocks 
       1% of 288 is `2.88` 
       This means  `± 2.88` blocks is the range where `Month` zoom button would be active.
       I think `±1` is more better than `± 2.88` when factoring precision loss.
       Also because integer division just affects the decimal places, `±1` should be okay.

Reply:
> `± 2.88` blocks is the range where `Day` zoom button would be active

That actually sounds great, but you can set it tighter if you want. The way you have it now, you are setting a 0.000000064% margin on time-based data, so there a lot of wiggle room to add. For use of `Zoom` on charts that with 0 < x < 1, this will simply fail, which is not acceptable. `Zoom` must handle data in a scale independent way.

Reply:
>For use of Zoom on charts that with 0 < x < 1, this will simply fail, which is not acceptable.

What does "x" here mean? A chart with only one block to plot?

Reply:
`x` is whatever the x-axis is. `Zoom` doesn't know or need to know what a block. It should handle fractional x values identically to integral x values.

Reply:
An error of `±0.99` should be better than ±1 since we don't have a 0.99 block (no such a counting number). On the time-based data this error has no much effect since its in microseconds and an error of ±0.99 microseconds is so small to be even noticed.

Reply:
It's fine for now.